src/call.js

```diff
@@ -1,18 +1,40 @@
-var net = require('net');
+"use strict"
 
-module.exports = function(name, args, callback) {
-  var client = net.createConnection({host: name, port: 9999}, function() {
-    client.end(JSON.stringify(args));
+const net = require('net');
+
+let call = (name, args, callback) => {
+  var client = net.createConnection({host: name, port: process.env.funker_port || 9999}, () => {
+    let stringified = JSON.stringify(args);
+    client.end(stringified);
   });
-  var buf = '';
-  client.on('error', function(err) {
+
+  var buffer = new Buffer(0);
+  client.on('error', (err) => {
     callback(err);
   });
-  client.on('data', function(data) {
-    buf += data;
+
+  client.on('data', (data) => {
+    var tempBuffer = new Buffer(data.length);
+    tempBuffer.write(data.toString(), "utf8");
+    buffer = Buffer.concat([buffer, tempBuffer]);
   });
-  client.on('end', function() {
-    var returnValue = JSON.parse(buf);
-    callback(null, returnValue);
+
+  client.on('end', () => {
+    let buf = buffer.toString("utf8");
+    var returnValue;
+    var err;
+
+    try {
+      returnValue = JSON.parse(buf);
+    } catch(e) {
+      let thrown = e;
+      console.error(thrown);
+      // Silly hack to get an error to serialize.
```

Comment:
In the Python and Go versions, we don't catch any errors. I quite like the idea of having a standard form for errors, but perhaps we should define it upstream in the bfirsh/funker repo then implement across all languages.

My feeling is that we'll need some top-level object that defines whether it is successful or not, otherwise funker errors will collide with user-defined objects with an error key.